src/app/product/productList/productlist.tsx

```diff
@@ -0,0 +1,153 @@
+"use client";
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+import {
+  AlertDialog,
+  AlertDialogAction,
+  AlertDialogCancel,
+  AlertDialogContent,
+  AlertDialogDescription,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogTitle,
+} from "@/components/ui/alert-dialog";
+import {
+  Typography,
+  Stack,
+  IconButton,
+  CircularProgress,
+  Box,
+} from "@mui/material";
+import { useState } from "react";
+import EditIcon from "@mui/icons-material/Edit";
+import DeleteIcon from "@mui/icons-material/Delete";
+import { Button } from "@/components/ui/button";
+import { useRouter } from "next/navigation";
+import { deleteProductAction } from "@/app/actions/actions";
+import { useToast } from "@/components/ui/use-toast";
+import { Product, ProductTableProps } from "@/lib/types";
+
+export default function ProductTable({ initialProducts }: ProductTableProps) {
+  const { toast } = useToast();
+  const [loading, setLoading] = useState(false);
+  const [products, setProducts] = useState<Product[]>(initialProducts);
+  const [selectedProduct, setSelectedProduct] = useState<Product | null>(null);
+  const [isDeleteDialogOpen, setIsDeleteDialogOpen] = useState(false);
+  const router = useRouter();
+
+  const handleDeleteClick = (product: Product, event: React.MouseEvent) => {
+    event.stopPropagation();
+    setIsDeleteDialogOpen(true);
+    setSelectedProduct(product);
+  };
+
+  const handleDeleteConfirm = async () => {
+    if (selectedProduct) {
+      const { success, message } = await deleteProductAction(
+        selectedProduct.id
+      );
+      if (success) {
+        setProducts(
+          products.filter((product) => product.id !== selectedProduct.id)
+        );
+        toast({
+          variant: "default",
+          description: message,
+        });
+        setIsDeleteDialogOpen(false);
+      }
+      if (message && !success) {
+        toast({
+          variant: "destructive",
+          description: `${message}`,
+        });
+      }
+    }
+  };
+
+  if (loading) {
```

Comment:
is this needed is? I don't see loading set to true anywhere

Reply:
handled loading state in page.tsx so removing this loading state.